src/dotnet/APIView/APIViewWeb/Controllers/AutoReviewController.cs

```diff
@@ -209,18 +209,21 @@ public async Task<ActionResult> CreateApiReview(
                         // But any manual pipeline run at release time should compare against all approved revisions to ensure hotfix release doesn't have API change
                         // If review surface doesn't match with any approved revisions then we will create new revision if it doesn't match pending latest revision
 
+                        var packageSemVer = new AzureEngSemanticVersion(codeFile.PackageVersion, codeFile.Language);
+                        var considerPackageVersion = (!packageSemVer.IsPrerelease) ? true : false;
+
                         if (compareAllRevisions)
                         {
                             foreach (var approvedAPIRevision in automaticRevisions.Where(r => r.IsApproved))
                             {
-                                if (await _apiRevisionsManager.AreAPIRevisionsTheSame(approvedAPIRevision, renderedCodeFile))
+                                if (await _apiRevisionsManager.AreAPIRevisionsTheSame(approvedAPIRevision, renderedCodeFile, considerPackageVersion))
                                 {
                                     return (review, approvedAPIRevision);
                                 }
                             }
                         }
 
-                        if (await _apiRevisionsManager.AreAPIRevisionsTheSame(latestAutomaticAPIRevision, renderedCodeFile))
+                        if (await _apiRevisionsManager.AreAPIRevisionsTheSame(latestAutomaticAPIRevision, renderedCodeFile, considerPackageVersion))
```

Comment:
We should also copy the approval status if a there is a matching revision with same exact api surface but different version so architects don't have to approve when just a version is changed.

Reply:
That would still happen https://github.com/chidozieononiwu/azure-sdk-tools/blob/ebf989d4b13b1ea0404afa5317997eac2c0f5039/src/dotnet/APIView/APIViewWeb/Controllers/AutoReviewController.cs#L250